### PR TITLE
Index broadcast

### PR DESCRIPTION
**Pull request by ngimel**
Fixes minimum repro from #623

### CODE REVIEW

File: torchinductor/lowering.py [hunk 1]
```diff
@@ -1337,12 +1337,35 @@ def fn(idx):
     )
 
 
+def check_and_broadcast_indices(indices):
+    print(indices)
```
Comment (jansel) [inline torchinductor/lowering.py:1341]: remove prints

File: torchinductor/lowering.py [hunk 2]
```diff
@@ -1337,12 +1337,35 @@ def fn(idx):
...
+    print([i.get_dtype() for i in indices if i is not None])
```
Comment (ngimel) [inline torchinductor/lowering.py:1342]:
```suggestion
```

**Review by jansel — APPROVED** [review]

**Comment by Chillee** [comment]
Does `vision_maskrcnn` work now btw?

**Comment by ngimel** [comment]
I don't know, #623 didn't have repro command. I suspect this PR breaks more than it fixes, lol, because indices are so often not int64 due to other bugs.

**Pull request merged**